python-threatexchange/threatexchange/cli/hash_cmd.py

```diff
@@ -99,10 +112,31 @@ def execute(self, settings: CLISettings) -> None:
                     f"{self.signal_type.get_name()} "
                     f"does not apply to {self.content_type.get_name()}"
                 )
+
             hashers = [self.signal_type]  # type: ignore  # can't detect intersection types
 
+        if not self.rotations:
+            for file in self.files:
+                for hasher in hashers:
+                    hash_str = hasher.hash_from_file(file)
+                    if hash_str:
+                        print(hasher.get_name(), hash_str)
+            return
+
+        if not issubclass(self.content_type, PhotoContent):
+            raise CommandError(
+                "--rotations flag is only available for Photo content type", 2
+            )
+
         for file in self.files:
-            for hasher in hashers:
-                hash_str = hasher.hash_from_file(file)
-                if hash_str:
-                    print(hasher.get_name(), hash_str)
+            with open(file, "rb") as f:
+                image_bytes = f.read()
+                rotated_images = PhotoContent.all_simple_rotations(image_bytes)
+                for _, rotated_bytes in rotated_images.items():
+                    with tempfile.NamedTemporaryFile() as temp_file:  # Create a temporary file to hold the byte data
+                        temp_file.write(rotated_bytes)
+                        temp_file_path = pathlib.Path(temp_file.name)
+                        for hasher in hashers:
+                            hash_str = hasher.hash_from_file(temp_file_path)
+                            if hash_str:
+                                print(hasher.get_name(), hash_str)
```

Comment:
blocking: Suggest we should change the output when using --rotations to also include the rotation

Suggest we change it to 

Before:
```
pdq accb6d39648035f8125c8ce6ba65007de7b54c67a2d93ef7b8f33b0611306715 
```
After:
```
original pdq accb6d39648035f8125c8ce6ba65007de7b54c67a2d93ef7b8f33b0611306715
```

This makes it not naively deserializable anymore (without using something like `cut` to manipulate the output, but I think it makes it better for human reading).

What do you think?

Reply:
That makes sense. I would also prefer it to output the rotation type to know what rotation matches with which pdq